pkg/registry/registrytest/service.go

```diff
@@ -45,10 +45,23 @@ func (r *ServiceRegistry) ListServices(ctx api.Context) (*api.ServiceList, error
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	// Return by copy to avoid data races
+	ns, _ := api.NamespaceFrom(ctx)
+
+	// Copy metadata from internal list into result
 	res := new(api.ServiceList)
-	*res = r.List
-	res.Items = append([]api.Service{}, r.List.Items...)
+	res.TypeMeta = r.List.TypeMeta
+	res.ListMeta = r.List.ListMeta
+
+	if ns != "" {
```

Comment:
make this api.NamespaceAll so I do not miss this when we make * be what "" is today.